openedx/core/djangoapps/credit/tests/test_api.py

```diff
@@ -181,6 +181,40 @@ def test_requirements_to_disable(self):
         requirements_to_disabled = _get_requirements_to_disable(old_requirements, requirements)
         self.assertEqual(len(requirements_to_disabled), 0)
 
+    def test_set_credit_requirement_status(self):
+        self.add_credit_course()
+        requirements = [
+            {
+                "namespace": "grade",
+                "name": "grade",
+                "display_name": "Grade",
+                "criteria": {
+                    "min_grade": 0.8
+                }
+            },
+            {
+                "namespace": "reverification",
+                "name": "i4x://edX/DemoX/edx-reverification-block/assessment_uuid",
+                "display_name": "Assessment 1",
+                "criteria": {}
+            }
+        ]
+
+        set_credit_requirements(self.course_key, requirements)
+        course_requirements = CreditRequirement.get_course_requirements(self.course_key)
+        self.assertEqual(len(course_requirements), 2)
+
+        requirement = CreditRequirement.get_course_requirement(self.course_key, "grade", "grade")
```

Comment:
If you keep the `get_credit_requirement` API function around, you should use it here instead of calling the method directly on the model. These are tests of the API, not the model.

That said, I think what you're doing here indicates that the API functions added in this PR are unnecessary indirection; indirection is not abstraction. You've already made `get_course_requirement` and `add_or_update_requirement_status` class methods. Why wrap them in a function call that adds nothing of value?